hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/merkle/disk/OnDiskKey.java

```diff
@@ -96,20 +95,26 @@ public void serialize(@NonNull final SerializableDataOutputStream out) throws IO
 
     @Override
     public void serialize(@NonNull final ByteBuffer byteBuffer) throws IOException {
+        throw new UnsupportedOperationException("Should never be called");
+        /*
         final var output = BufferedData.wrap(byteBuffer);
         output.skip(4);
         codec.write(key, output);
         final var pos = output.position();
         output.position(0);
         output.writeInt((int) pos - 4);
         output.position(pos);
+        */
```

Comment:
Yes, it can be removed. I wasn't sure about it when I started working on this fix, that's why just temporarily commented it out.

Here is the full story. First, `VirtualKey` and `VirtualValue` platform classes only had methods to serialize to and deserialize from streams. These methods are still there, they are used to write keys/values from virtual node cache to snapshots. Then at some point I added similar methods to work with byte buffers, because at the JasperDB / MerkleDb level all serialization was done using byte buffers. Now with switch to PBJ, byte buffers won't be used any more, but `BufferedData` will be used instead. At the same time, having these methods directly in key / value classes doesn't make sense, as we have `KeySerializer` and `ValueSerializer` interfaces specifically for key and value serialization needs. That's why I am going to drop the methods from `VirtualKey` and `VirtualValue` altogether, which will be possible after JasperDB is removed. Here in `OnDiskKey` and `OnDiskValue`, I assume these classes will not be used until we switch to PBJ in MerkleDb.

While I was writing the paragraph above, I found that even today, before switching to PBJ, MerkleDb doesn't use `serialize()` / `deserialize()` on virtual keys or values. It uses the corresponding methods in `KeySerializer` and `ValueSerializer` interfaces. That's why it's safe to throw UnsupportedOperationException here, and eventually these methods will be removed completely.

Reply:
I'll remove the commented code.